tests/registrar_test.php

```diff
@@ -199,4 +199,21 @@ public function testRegisterDoesNotCallUnregisterable()
 
         $this->assertEquals(0, $called);
     }
+
+    public function testRegisterMixedNamespaces()
+    {
+        $registrar = \Dxw\Iguana\Registrar::getSingleton();
+
+        $registrar->setNamespace('meow');
+        $registrar->addInstance('MyRegisterable', new \MyRegisterable());
+        $registrar->setNamespace('woof');
+        $registrar->addInstance('MyRegisterable', new \MyRegisterable());
+
+        global $called;
+        $called = 0;
+
+        $registrar->register();
+
+        $this->assertEquals(1, $called);
```

Comment:
Would it be worth tightening up this test so that we can tell that the class in the correct namespace has been called, rather than just keeping the count? Maybe `$called` could be an associative array that keeps track of how many `register()` methods have been called in each namespace?